File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.py
class Solution:
    def average(self, salary: List[int]) -> float:
        max_salary = -inf
        min_salary = inf
        total = 0
        for s in salary:
            max_salary = max(max_salary, s)
            min_salary = min(min_salary, s)
            total += s
        total -= max_salary + min_salary
        return total / (len(salary) - 2)

File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.java
class Solution {
    public double average(int[] salary) {
        int min = Integer.MAX_VALUE, max = Integer.MIN_VALUE, total = 0;
        for(int s : salary){
            min = Math.min(min, s);
            max = Math.max(max, s);
            total += s;
        }
        return (total - max - min) / (salary.length - 2.0);
    }
}